<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#EDA" data-toc-modified-id="EDA-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>EDA</a></span><ul class="toc-item"><li><span><a href="#Preprocessing" data-toc-modified-id="Preprocessing-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Preprocessing</a></span></li></ul></li><li><span><a href="#Subset-Check" data-toc-modified-id="Subset-Check-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Subset Check</a></span><ul class="toc-item"><li><span><a href="#View-all-the-apps-listed-before-the-&quot;Mandala-Coloring-Book&quot;" data-toc-modified-id="View-all-the-apps-listed-before-the-&quot;Mandala-Coloring-Book&quot;-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>View all the apps listed before the "Mandala Coloring Book"</a></span><ul class="toc-item"><li><span><a href="#Option-1" data-toc-modified-id="Option-1-2.1.1"><span class="toc-item-num">2.1.1&nbsp;&nbsp;</span>Option 1</a></span></li><li><span><a href="#Option-2" data-toc-modified-id="Option-2-2.1.2"><span class="toc-item-num">2.1.2&nbsp;&nbsp;</span>Option 2</a></span></li></ul></li><li><span><a href="#List-only-the-ratings-of-all-the-apps--between-'Paper-flowers-instructions'-and-'Text-on-Photo---Fonteee'" data-toc-modified-id="List-only-the-ratings-of-all-the-apps--between-'Paper-flowers-instructions'-and-'Text-on-Photo---Fonteee'-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>List only the ratings of all the apps  between 'Paper flowers instructions' and 'Text on Photo - Fonteee'</a></span></li><li><span><a href="#List-all-the-columns-bewteen-'Size'-and-'Content-Rating'" data-toc-modified-id="List-all-the-columns-bewteen-'Size'-and-'Content-Rating'-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>List all the columns bewteen 'Size' and 'Content Rating'</a></span></li><li><span><a href="#What-is-the-average-rating-of-apps-in-the-store?" data-toc-modified-id="What-is-the-average-rating-of-apps-in-the-store?-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>What is the average rating of apps in the store?</a></span></li><li><span><a href="#What-is-the-ratio-of-free-vs-paid?" data-toc-modified-id="What-is-the-ratio-of-free-vs-paid?-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>What is the ratio of free vs paid?</a></span></li><li><span><a href="#What-is-the-highest-amount-of-Reviews-an-app-received?-And-the-lowest?" data-toc-modified-id="What-is-the-highest-amount-of-Reviews-an-app-received?-And-the-lowest?-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>What is the highest amount of Reviews an app received? And the lowest?</a></span></li><li><span><a href="#What-are-the-ten-most-recently-updated-apps?" data-toc-modified-id="What-are-the-ten-most-recently-updated-apps?-2.7"><span class="toc-item-num">2.7&nbsp;&nbsp;</span>What are the ten most recently updated apps?</a></span></li><li><span><a href="#Find-apps,-that-is-in-the-top-25%-of-apps-based-on-ratings-and-were-updted-in-the-last-two-months-of-dataset" data-toc-modified-id="Find-apps,-that-is-in-the-top-25%-of-apps-based-on-ratings-and-were-updted-in-the-last-two-months-of-dataset-2.8"><span class="toc-item-num">2.8&nbsp;&nbsp;</span>Find apps, that is in the top 25% of apps based on ratings and were updted in the last two months of dataset</a></span></li><li><span><a href="#Which-app-not-to-download?---Find-an-app-which-takes-the-most-space-on-your-phonw,-but-is-rated-poorly-(define-poorly)" data-toc-modified-id="Which-app-not-to-download?---Find-an-app-which-takes-the-most-space-on-your-phonw,-but-is-rated-poorly-(define-poorly)-2.9"><span class="toc-item-num">2.9&nbsp;&nbsp;</span>Which app not to download? - Find an app which takes the most space on your phonw, but is rated poorly (define poorly)</a></span></li></ul></li></ul></div>

In [1]:
import os
import math

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
def load_data(data_path):
    return pd.read_csv(data_path)

In [3]:
"""
Initialization 
"""
data_dir = 'google-play-store-apps'
data_file1 = 'googleplaystore.csv'
data_file2 = 'googleplaystore_user_reviews.csv'
data_path1 = os.path.join(os.getcwd(), 'data', data_dir, data_file1)
data_path2 = os.path.join(os.getcwd(), 'data', data_dir, data_file2)

In [4]:
df_store = load_data(data_path1)

In [5]:
df_review = load_data(data_path2)

### EDA

In [6]:
df_store.shape

(10841, 13)

In [7]:
df_store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [8]:
df_review.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
App                       64295 non-null object
Translated_Review         37427 non-null object
Sentiment                 37432 non-null object
Sentiment_Polarity        37432 non-null float64
Sentiment_Subjectivity    37432 non-null float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


In [9]:
df_store['Rating'].describe()

count    9367.000000
mean        4.193338
std         0.537431
min         1.000000
25%         4.000000
50%         4.300000
75%         4.500000
max        19.000000
Name: Rating, dtype: float64

#### Preprocessing

In [10]:
# Convert data with Rating value greater than 5 to None
df_store = df_store[(df_store['Rating'] <= 5) | (df_store['Rating'].isnull())]

In [11]:
df_store['Reviews'] = df_store['Reviews'].astype(int)

In [12]:
df_store['Rating'].shape

(10840,)

In [13]:
def convert_size2MB(value):
    val = str(value)
    if 'Varies with device' == val:
        return None
    elif 'M' in val:
        if ',' in val:
            val = val.replace(',', '')
        return val.replace('M', '')
    elif 'k' in val:
        return float(val.replace('k', '')) / 1000
    else:
        return value

# Convert size to MB
df_store['Size'] = df_store['Size'].apply(convert_size2MB).astype(float)

In [14]:
df_review.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [15]:
df_store['App'].value_counts()

ROBLOX                                                9
CBS Sports App - Scores, News, Stats & Watch Live     8
8 Ball Pool                                           7
ESPN                                                  7
Duolingo: Learn Languages Free                        7
Candy Crush Saga                                      7
Bubble Shooter                                        6
Nick                                                  6
Sniper 3D Gun Shooter: Free Shooting Games - FPS      6
Zombie Catchers                                       6
Bleacher Report: sports news, scores, & highlights    6
Temple Run 2                                          6
slither.io                                            6
Bowmasters                                            6
Helix Jump                                            6
Subway Surfers                                        6
Wish - Shopping Made Fun                              5
Skyscanner                                      

In [16]:
df_review['App'].value_counts()

CBS Sports App - Scores, News, Stats & Watch Live     320
Angry Birds Classic                                   320
Bowmasters                                            320
8 Ball Pool                                           300
Helix Jump                                            300
Bubble Shooter                                        260
Calorie Counter - MyFitnessPal                        260
Candy Crush Saga                                      240
DEAD TARGET: FPS Zombie Apocalypse Survival Games     240
ESPN                                                  240
Garena Free Fire                                      240
Duolingo: Learn Languages Free                        240
Flow Free                                             220
Google Photos                                         220
Hangouts                                              220
Hill Climb Racing                                     220
Block Puzzle                                          220
Farm Heroes Sa

### Subset Check

In [17]:
df_store[df_store['App'] == 'Pixel Draw - Number Art Coloring Book']

App        Category  Rating  Reviews  \
4  Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3      967   

   Size  Installs  Type Price Content Rating                   Genres  \
4   2.8  100,000+  Free     0       Everyone  Art & Design;Creativity   

    Last Updated Current Ver Android Ver  
4  June 20, 2018         1.1  4.4 and up

In [18]:
df_store[df_store['App'].isin(['Pixel Draw - Number Art Coloring Book', 
                             'Beauty Selfie Camera',
                             'BestCam Selfie-selfie, beauty camera, photo editor'])]

App        Category  \
4                Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN   
101  BestCam Selfie-selfie, beauty camera, photo ed...          BEAUTY   
103                               Beauty Selfie Camera          BEAUTY   

     Rating  Reviews  Size  Installs  Type Price Content Rating  \
4       4.3      967   2.8  100,000+  Free     0       Everyone   
101     3.9     1739  21.0  500,000+  Free     0       Everyone   
103     4.2     2225  52.0  500,000+  Free     0       Everyone   

                      Genres       Last Updated Current Ver   Android Ver  
4    Art & Design;Creativity      June 20, 2018         1.1    4.4 and up  
101                   Beauty      July 12, 2018       1.0.6  4.0.3 and up  
103                   Beauty  February 28, 2018         1.6    4.1 and up

#### View all the apps listed before the "Mandala Coloring Book"

##### Option 1

In [19]:
# df_store.index = df_store['App']
# df_store[:'Mandala Coloring Book']

# df_store.index = [x for x in range(df_store.shape[0])]

##### Option 2

In [20]:
df_store.loc[:'Mandala Coloring Book']

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                             Paper flowers instructions       ART_AND_DESIGN   
6                Smoke Effect Photo Maker - Smoke Editor       ART_AND_DESIGN   
7                                       Infinite Painter       ART_AND_DESIGN   
8                                   Garden Coloring Book       ART_AND_DESIGN   
9                          Kids Paint Free - Drawing Fun       ART_AND_DESIGN   
10                               Text on Photo - Fonteee       ART_AND_DESIGN   
11               Name Art Photo Editor - Focus n Filters       ART_AND_DESIGN   
12                        Tattoo Name On My Photo Editor       ART_AND_DESIGN   
13                                 Mandala Coloring Book       ART_AND_DESIGN   
14       3D Color Pixel by Number - Sandbox Art Coloring       ART_AND_DESIGN   
15                       Learn To Draw Kawaii Characters       ART_AND_DESIGN   
16          Photo Designer - Write your name with shapes       ART_AND_DESIGN   
17                              350 Diy Room Decor Ideas       ART_AND_DESIGN   
18                         FlipaClip - Cartoon animation       ART_AND_DESIGN   
19                                          ibis Paint X       ART_AND_DESIGN   
20                           Logo Maker - Small Business       ART_AND_DESIGN   
21             Boys Photo Editor - Six Pack & Men's Suit       ART_AND_DESIGN   
22               Superheroes Wallpapers | 4K Backgrounds       ART_AND_DESIGN   
23                                Mcqueen Coloring pages       ART_AND_DESIGN   
24                           HD Mickey Minnie Wallpapers       ART_AND_DESIGN   
25                            Harley Quinn wallpapers HD       ART_AND_DESIGN   
26                         Colorfit - Drawing & Coloring       ART_AND_DESIGN   
27                                 Animated Photo Editor       ART_AND_DESIGN   
28                                 Pencil Sketch Drawing       ART_AND_DESIGN   
29                       Easy Realistic Drawing Tutorial       ART_AND_DESIGN   
...                                                  ...                  ...   
10811                                        FR Plus 1.6    AUTO_AND_VEHICLES   
10812                                      Fr Agnel Pune               FAMILY   
10813                                     DICT.fr Mobile             BUSINESS   
10814                               FR: My Secret Pets!                FAMILY   
10815                          Golden Dictionary (FR-AR)  BOOKS_AND_REFERENCE   
10816                                 FieldBi FR Offline             BUSINESS   
10817                               HTC Sense Input - FR                TOOLS   
10818                               Gold Quote - Gold.fr              FINANCE   
10819                                          Fanfic-FR  BOOKS_AND_REFERENCE   
10820                                    Fr. Daoud Lamei               FAMILY   
10821                                            Poop FR               FAMILY   
10822                                          PLMGSS FR         PRODUCTIVITY   
10823                                       List iptv FR        VIDEO_PLAYERS   
10824                                          Cardio-FR              MEDICAL   
10825                                 Naruto & Boruto FR               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10827                                 Fr Agnel Ambarnath               FAMILY   
10828                            Manga-FR - Anime Vostfr               COMICS   
108

In [21]:
# for app in df_store['App'].value_counts().index:
#     print(app)

#### List only the ratings of all the apps  between 'Paper flowers instructions' and 'Text on Photo - Fonteee'

In [22]:
df_store[df_store['App'] == 'Paper flowers instructions']

App        Category  Rating  Reviews  Size Installs  \
5  Paper flowers instructions  ART_AND_DESIGN     4.4      167   5.6  50,000+   

   Type Price Content Rating        Genres    Last Updated Current Ver  \
5  Free     0       Everyone  Art & Design  March 26, 2017         1.0   

  Android Ver  
5  2.3 and up

In [23]:
df_store[df_store['App'] == 'Text on Photo - Fonteee']

App        Category  Rating  Reviews  Size  \
10  Text on Photo - Fonteee  ART_AND_DESIGN     4.4    13880  28.0   

      Installs  Type Price Content Rating        Genres      Last Updated  \
10  1,000,000+  Free     0       Everyone  Art & Design  October 27, 2017   

   Current Ver Android Ver  
10       1.0.4  4.1 and up

In [24]:
app_index1 = df_store[df_store['App'] == 'Paper flowers instructions'].index.values[0]
app_index2 = df_store[df_store['App'] == 'Text on Photo - Fonteee'].index.values[0]

df_store['Rating'].loc[app_index1 : app_index2]

5     4.4
6     3.8
7     4.1
8     4.4
9     4.7
10    4.4
Name: Rating, dtype: float64

#### List all the columns bewteen 'Size' and 'Content Rating'

In [25]:
df_store.loc[:, 'Size':'Content Rating']

Size     Installs  Type Price Content Rating
0      19.000      10,000+  Free     0       Everyone
1      14.000     500,000+  Free     0       Everyone
2       8.700   5,000,000+  Free     0       Everyone
3      25.000  50,000,000+  Free     0           Teen
4       2.800     100,000+  Free     0       Everyone
5       5.600      50,000+  Free     0       Everyone
6      19.000      50,000+  Free     0       Everyone
7      29.000   1,000,000+  Free     0       Everyone
8      33.000   1,000,000+  Free     0       Everyone
9       3.100      10,000+  Free     0       Everyone
10     28.000   1,000,000+  Free     0       Everyone
11     12.000   1,000,000+  Free     0       Everyone
12     20.000  10,000,000+  Free     0           Teen
13     21.000     100,000+  Free     0       Everyone
14     37.000     100,000+  Free     0       Everyone
15      2.700       5,000+  Free     0       Everyone
16      5.500     500,000+  Free     0       Everyone
17     17.000      10,000+  Free     0       Everyone
18     39.000   5,000,000+  Free     0       Everyone
19     31.000  10,000,000+  Free     0       Everyone
20     14.000     100,000+  Free     0       Everyone
21     12.000     100,000+  Free     0       Everyone
22      4.200     500,000+  Free     0   Everyone 10+
23      7.000     100,000+  Free     0       Everyone
24     23.000      50,000+  Free     0       Everyone
25      6.000      10,000+  Free     0       Everyone
26     25.000     500,000+  Free     0       Everyone
27      6.100     100,000+  Free     0       Everyone
28      4.600      10,000+  Free     0       Everyone
29      4.200     100,000+  Free     0       Everyone
...       ...          ...   ...   ...            ...
10811   3.900         100+  Free     0       Everyone
10812  13.000       1,000+  Free     0       Everyone
10813   2.700      10,000+  Free     0       Everyone
10814  31.000      50,000+  Free     0           Teen
10815   4.900     500,000+  Free     0       Everyone
10816   6.800         100+  Free     0       Everyone
10817   8.000     100,000+  Free     0       Everyone
10818   1.500      10,000+  Free     0       Everyone
10819   3.600       5,000+  Free     0           Teen
10820   8.600       1,000+  Free     0           Teen
10821   2.500          50+  Free     0       Everyone
10822   3.100          10+  Free     0       Everyone
10823   2.900         100+  Free     0       Everyone
10824  82.000      10,000+  Free     0       Everyone
10825   7.700         100+  Free     0           Teen
10826     NaN   5,000,000+  Free     0     Mature 17+
10827  13.000       5,000+  Free     0       Everyone
10828  13.000      10,000+  Free     0       Everyone
10829   7.400      10,000+  Free     0       Everyone
10830   2.300     100,000+  Free     0       Everyone
10831   9.800       5,000+  Free     0       Everyone
10832   0.582     100,000+  Free     0       Everyone
10833   0.619       1,000+  Free     0       Everyone
10834   2.600         500+  Free     0       Everyone
10835   9.600          10+  Free     0       Everyone
10836  53.000       5,000+  Free     0       Everyone
10837   3.600         100+  Free     0       Everyone
10838   9.500       1,000+  Free     0       Everyone
10839     NaN       1,000+  Free     0     Mature 17+
10840  19.000  10,000,000+  Free     0       Everyone

[10840 rows x 5 columns]

#### What is the average rating of apps in the store?

In [26]:
df_store['Rating'].mean()

4.191757420456978

In [27]:
df_store['Rating'][~df_store['Rating'].isnull()].mean()

4.191757420456978

#### What is the ratio of free vs paid?

In [28]:
df_store['Type'].value_counts()

Free    10039
Paid      800
Name: Type, dtype: int64

#### What is the highest amount of Reviews an app received? And the lowest?

In [29]:
# The highest amount of Reviews
display(df_store.sort_values(by='Reviews', ascending= False).iloc[0])

App                         Facebook
Category                      SOCIAL
Rating                           4.1
Reviews                     78158306
Size                             NaN
Installs              1,000,000,000+
Type                            Free
Price                              0
Content Rating                  Teen
Genres                        Social
Last Updated          August 3, 2018
Current Ver       Varies with device
Android Ver       Varies with device
Name: 2544, dtype: object

In [30]:
# The lowest amount of Reviews
df_store[df_store['Reviews'] == '0']

C:\Users\Yuan_Liao\AppData\Local\Continuum\anaconda3\envs\dsa\lib\site-packages\pandas\core\ops.py:1649: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = method(y)


Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

#### What are the ten most recently updated apps?

In [31]:
df_store['Last_Updated_dt'] = pd.to_datetime(df_store['Last Updated'], format= '%B %d, %Y')
df_store.sort_values(by='Last_Updated_dt', ascending= False)[:10]

App            Category  \
10712                              Lalafo Pulsuz Elanlar            SHOPPING   
10209       Video Downloader For FB: Save FB Videos 2018               TOOLS   
10408                            Shoot Hunter-Gun Killer                GAME   
10718                                         BankNordik             FINANCE   
10760                                    Fast Tract Diet  HEALTH_AND_FITNESS   
10502                         Fun Kid Racing - Motocross              FAMILY   
10301  FD Calculator (EMI, SIP, RD & Loan Eligilibility)             FINANCE   
10373                                             FG VOC            BUSINESS   
10790                             HipChat - beta version       COMMUNICATION   
8749                                Monster Fishing 2018              SPORTS   

       Rating  Reviews  Size     Installs  Type  Price Content Rating  \
10712     4.4    61392   NaN   1,000,000+  Free      0       Everyone   
10209     4.3      434   3.6      50,000+  Free      0       Everyone   
10408     4.3   320334  27.0  50,000,000+  Free      0           Teen   
10718     3.9       28  15.0       5,000+  Free      0       Everyone   
10760     4.4       35   2.4       1,000+  Paid  $7.99       Everyone   
10502     4.1    59768   NaN  10,000,000+  Free      0       Everyone   
10301     5.0      104   2.3       1,000+  Free      0       Everyone   
10373     4.8        6   7.7       1,000+  Free      0       Everyone   
10790     4.1     1035  20.0      50,000+  Free      0       Everyone   
8749      4.4    32881  76.0   1,000,000+  Free      0       Everyone   

                          Genres    Last Updated         Current Ver  \
10712                   Shopping  August 8, 2018  Varies with device   
10209                      Tools  August 8, 2018               1.1.9   
10408                     Action  August 8, 2018               1.1.2   
10718                    Finance  August 8, 2018               7.3.2   
10760           Health & Fitness  August 8, 2018               1.9.3   
10502  Racing;Action & Adventure  August 7, 2018                3.53   
10301                    Finance  August 7, 2018               2.1.0   
10373                   Business  August 7, 2018               2.1.5   
10790              Communication  August 7, 2018            3.20.001   
8749                      Sports  August 7, 2018              0.0.80   

              Android Ver Last_Updated_dt  
10712  Varies with device      2018-08-08  
10209        4.0.3 and up      2018-08-08  
10408          4.1 and up      2018-08-08  
10718          5.0 and up      2018-08-08  
10760          4.2 and up      2018-08-08  
10502          4.2 and up      2018-08-07  
10301          4.1 and up      2018-08-07  
10373          4.1 and up      2018-08-07  
10790          4.1 and up      2018-08-07  
8749           4.1 and up      2018-08-07

#### Find apps, that is in the top 25% of apps based on ratings and were updted in the last two months of dataset

In [32]:
rating_score = sorted(df_store['Rating'].dropna().unique())
rating_top25 = rating_score[math.floor((len(rating_score) * 0.75))]

# Top 25% apps based on ratings
df_store_rating_top25 = df_store[df_store['Rating'] > rating_top25]

In [33]:
last_2_month = df_store.sort_values(by='Last_Updated_dt', 
                                    ascending = False)['Last_Updated_dt'].iloc[0] \
                                    - pd.DateOffset(months = 2)
last_2_month.to_period('M')
# Last 2 months (based on dataset date)
df_last_2_months = df_store[df_store['Last_Updated_dt'].dt.to_period('M') > last_2_month.to_period('M')]

In [35]:
# Result 
np.intersect1d(df_store_rating_top25['App'], df_last_2_months['App'])

array(['+Download 4 Instagram Twitter', '/u/app', '058.ba', ...,
       '乗換NAVITIME\u3000Timetable & Route Search in Japan Tokyo',
       '💘 WhatsLov: Smileys of love, stickers and GIF',
       '🔥 Football Wallpapers 4K | Full HD Backgrounds 😍'], dtype=object)

#### Which app not to download? - Find an app which takes the most space on your phonw, but is rated poorly (define poorly)

In [843]:
# for app in df_store['App']:
#     print(app)

In [40]:
size_max = df_store[~df_store['Size'].isnull()].sort_values(by='Size', 
                                                            ascending= False)['Size'].values[0] 
df_store_size_max = df_store[df_store['Size'] == size_max]

In [41]:
df_store_size_max.sort_values(by='Rating', ascending= True)

App            Category  \
4690                                        Vi Trainer  HEALTH_AND_FITNESS   
8409  Car Crash III Beam DH Real Damage Simulator 2018                GAME   
1565                    Talking Babsy Baby: Baby Games           LIFESTYLE   
5530                       The Walking Dead: Our World                GAME   
5862                             Miami crime simulator                GAME   
5427                                   Ultimate Tennis              SPORTS   
5865                      Gangster Town: Vice District              FAMILY   
9170                     Stickman Legends: Shadow Wars                GAME   
1080                                         Post Bank             FINANCE   
1758                            Hungry Shark Evolution                GAME   
1793                 Mini Golf King - Multiplayer Game                GAME   
1988                            Hungry Shark Evolution                GAME   
3973                            Hungry Shark Evolution                GAME   
7404                                   SimCity BuildIt              FAMILY   
8847                        Draft Simulator for FUT 18              SPORTS   
2299                              Navi Radiography Pro             MEDICAL   

      Rating  Reviews   Size      Installs  Type   Price Content Rating  \
4690     3.6      124  100.0        5,000+  Free       0       Everyone   
8409     3.6      151  100.0       10,000+  Free       0       Everyone   
1565     4.0   140995  100.0   10,000,000+  Free       0       Everyone   
5530     4.0    22435  100.0    1,000,000+  Free       0           Teen   
5862     4.0   254518  100.0   10,000,000+  Free       0     Mature 17+   
5427     4.3   183004  100.0   10,000,000+  Free       0       Everyone   
5865     4.3    65146  100.0   10,000,000+  Free       0     Mature 17+   
9170     4.4    38419  100.0    1,000,000+  Paid   $0.99   Everyone 10+   
1080     4.5    60449  100.0    1,000,000+  Free       0       Everyone   
1758     4.5  6074334  100.0  100,000,000+  Free       0           Teen   
1793     4.5   531458  100.0    5,000,000+  Free       0       Everyone   
1988     4.5  6074627  100.0  100,000,000+  Free       0           Teen   
3973     4.5  6071542  100.0  100,000,000+  Free       0           Teen   
7404     4.5  4218587  100.0   50,000,000+  Free       0   Everyone 10+   
8847     4.6   162933  100.0    5,000,000+  Free       0       Everyone   
2299     4.7       11  100.0          500+  Paid  $15.99       Everyone   

                      Genres      Last Updated   Current Ver   Android Ver  \
4690        Health & Fitness    August 2, 2018      2.0.0.11    5.0 and up   
8409                  Racing      May 20, 2018          1.03    4.1 and up   
1565  Lifestyle;Pretend Play     July 16, 2018           9.0    4.0 and up   
5530                  Action    August 1, 2018       1.1.0.3    5.0 and up   
5862                  Action      July 9, 2018           2.0    4.0 and up   
5427                  Sports     July 19, 2018     2.29.3118  4.0.3 and up   
5865              Simulation      May 31, 2018           2.1    4.0 and up   
9170                  Action    August 3, 2018        2.3.24    4.1 and up   
1080                 Finance     July 23, 2018        2.9.12    4.0 and up   
1758                  Arcade     July 25, 2018         6.0.0    4.1 and up   
1793                  Sports     July 20, 2018        3.04.1  4.0.3 and up   
1988                  Arcade     July 25, 2018         6.0.0    4.1 and up   
3973                  Arcade     July 25, 2018         6.0.0    4.1 and up   
7404              Simulation     June 19, 2018  1.23.3.75024    4.0 and up   
8847                  Sports      May 11, 2018          25.7    4.1 and up   
2299                 Medical  January 14, 2018         1.2.0  4.0.3 and up   

     Last_Updated_dt  
4690      2018-08-02  
8409      2018-05-20  
1565      2018-07-16  
5530      2018-08-01  
5862  

Considering ratings of App that belong to the 'HEALTH_AND_FITNESS' category and 'Health & Fitness' Genres, the highest rating reachs 5.0, and the mean rating is around 4.3, while the lowest one is only 1.4

In [54]:
rating_size_heads = np.array(['App', 'Rating', 'Size', 'Installs', 'Type', 
                              'Price', 'Category', 'Genres'])
df_store[(df_store['Category'] == 'HEALTH_AND_FITNESS') &
         (df_store['Genres'] == 'Health & Fitness')][rating_size_heads]['Rating'].describe()

count    297.000000
mean       4.277104
std        0.617822
min        1.400000
25%        4.100000
50%        4.500000
75%        4.600000
max        5.000000
Name: Rating, dtype: float64

Considering ratings of App that belong to the 'GAME' category and 'Racing' Genres, the highest rating reachs 5.0, and the mean rating is around 4.1, but the lowest rating still has 3.3

In [55]:
df_store[(df_store['Category'] == 'GAME') &
         (df_store['Genres'] == 'Racing')][rating_size_heads]['Rating'].describe()

count    93.000000
mean      4.173118
std       0.327089
min       3.300000
25%       3.900000
50%       4.200000
75%       4.400000
max       5.000000
Name: Rating, dtype: float64

In [43]:
df_store_size_max[df_store_size_max['Genres'] == 'Health & Fitness']

App            Category  Rating  Reviews   Size Installs  Type  \
4690  Vi Trainer  HEALTH_AND_FITNESS     3.6      124  100.0   5,000+  Free   

     Price Content Rating            Genres    Last Updated Current Ver  \
4690     0       Everyone  Health & Fitness  August 2, 2018    2.0.0.11   

     Android Ver Last_Updated_dt  
4690  5.0 and up      2018-08-02

In the app store dataset, the largest application size is around **100MB**. <br>
And there are two APPs got the same lowest rating (3.6) among the 100MB size applications, one is health care related app, another one is a game. <br>
In order to decide which one is rated poorly and not to download:
- Both apps are free, and have thousand level installs. <br>
- For the app, Vi Trainer, because the range of HEALTH_AND_FITNESS category apps is from 1.4 to 5, and the mean rating is around 4.27. We here could consider this app with 3.6 rating is a not so bad app for HEALTH_AND_FITNESS. <br>
- For the app, Car Crash III Beam DH Real Damage Simulator 2018, which is a racing game. Although the mean rating over the similar games is 4.17, the lowest rating is 3.3. A racing game only got 3.6 rating, and only had 10 thousand installs. We can not define this as a good application. <br> <br>
So DO NO INSTALL Car Crash III Beam DH Real Damage Simulator 2018.

In [477]:
"""
Create a reviews reliability of reviews index, which takes into account,
how many reviews does the app have and how high the rating is.
What metric do you take? Create as a new field.
"""

'\nCreate a reviews reliability of reviews index, which takes into account,\nhow many reviews does the app have and how high the rating is.\nWhat metric do you take? Create as a new field.\n'

In [478]:
df_review

App  \
0            10 Best Foods for You   
1            10 Best Foods for You   
2            10 Best Foods for You   
3            10 Best Foods for You   
4            10 Best Foods for You   
5            10 Best Foods for You   
6            10 Best Foods for You   
7            10 Best Foods for You   
8            10 Best Foods for You   
9            10 Best Foods for You   
10           10 Best Foods for You   
11           10 Best Foods for You   
12           10 Best Foods for You   
13           10 Best Foods for You   
14           10 Best Foods for You   
15           10 Best Foods for You   
16           10 Best Foods for You   
17           10 Best Foods for You   
18           10 Best Foods for You   
19           10 Best Foods for You   
20           10 Best Foods for You   
21           10 Best Foods for You   
22           10 Best Foods for You   
23           10 Best Foods for You   
24           10 Best Foods for You   
25           10 Best Foods for You   
26           10 Best Foods for You   
27           10 Best Foods for You   
28           10 Best Foods for You   
29           10 Best Foods for You   
...                            ...   
64265  Houzz Interior Design Ideas   
64266  Houzz Interior Design Ideas   
64267  Houzz Interior Design Ideas   
64268  Houzz Interior Design Ideas   
64269  Houzz Interior Design Ideas   
64270  Houzz Interior Design Ideas   
64271  Houzz Interior Design Ideas   
64272  Houzz Interior Design Ideas   
64273  Houzz Interior Design Ideas   
64274  Houzz Interior Design Ideas   
64275  Houzz Interior Design Ideas   
64276  Houzz Interior Design Ideas   
64277  Houzz Interior Design Ideas   
64278  Houzz Interior Design Ideas   
64279  Houzz Interior Design Ideas   
64280  Houzz Interior Design Ideas   
64281  Houzz Interior Design Ideas   
64282  Houzz Interior Design Ideas   
64283  Houzz Interior Design Ideas   
64284  Houzz Interior Design Ideas   
64285  Houzz Interior Design Ideas   
64286  Houzz Interior Design Ideas   
64287  Houzz Interior Design Ideas   
64288  Houzz Interior Design Ideas   
64289  Houzz Interior Design Ideas   
64290  Houzz Interior Design Ideas   
64291  Houzz Interior Design Ideas   
64292  Houzz Interior Design Ideas   
64293  Houzz Interior Design Ideas   
64294  Houzz Interior Design Ideas   

                                       Translated_Review Sentiment  \
0      I like eat delicious food. That's I'm cooking ...  Positive   
1        This help eating healthy exercise regular basis  Positive   
2                                                    NaN       NaN   
3             Works great especially going grocery store  Positive   
4                                           Best idea us  Positive   
5                                               Best way  Positive   
6                                                Amazing  Positive   
7                                                    NaN       NaN   
8                                   Looking forward app,   Neutral   
9                  It helpful site ! It help foods get !   Neutral   
10                                             good you.  Positive   
11     Useful information The amount spelling errors ...  Positive   
12     Thank you! Great app!! Add arthritis, eyes, im...  Positive   
13     Greatest ever Completely awesome maintain heal...  Positive   
14     Good health...... Good health first priority.....  Positive   
15                                                   NaN       NaN   
16     Health It's important world either life . thin...  Positive   
17     Mrs sunita bhati I thankful developers,to make...  Positive   
18     Very Useful in diabetes age 30. I need control...  Positive   
19                                    One greatest apps.  Positive   
20                                             good nice  Positive   
21                                 Healthy Really helped  Positive   
22                                            God health   Neutr